new/default-resource-name.md

```diff
@@ -0,0 +1,103 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Default Resource Name
+
+In which resources get names and DSL by default, unless you say you don't want one.
+
+## Motivation
+
+    As a Chef user,
+    I want convention to drive my DSL,
+    So that I can concentrate on my code instead of ceremony.
+
+## Specification
+
+When a user declares a resource class thus:
+
+```ruby
+class MyResource < Chef::Resource
+end
+```
+
+It will automatically gain a resource_name and DSL derived from the class:
+
+```ruby
+my_resource 'foo' do
+end
+```
+
+This is already (currently) the case for classes declared in Chef::Resource; this extends the rule to all classes anywhere and removes the special case.
+
+### resource_name creates automatic DSL
+
+When a class's resource_name is set, it will always get DSL will that resource name. Thus, this will work:
+
+```ruby
+class X < Chef::Resource
+  resource_name :my_resource
+end
+my_resource 'foo' do
+end
+```
+
+#### New resource classes' resource names are set automatically
+
+When a resource class is declared with an explicit name, its resource name is set automatically:
+
+```ruby
+class SuperAwesome < Chef::Resource
+end
+super_awesome 'foo' do
+end
+```
+
+This will *not* apply to classes manufactured using Class.new and assigned to constants or given names later.
+
+#### resource_name nil unsets resource name
+
+class.resource_name nil unsets a resource name (and prevents that name from appearing in DSL).
```

Comment:
What about using `false` instead of `nil` for unsetting?  I ask because we have run across a lot of provisioning use cases where we want to unset something (like an attribute/property) and `nil` already communicates 'don't change this' because `nil` is the default.